cpp/unreal_projects/SpearSim/Python/validate_spear_scene.py

```diff
@@ -0,0 +1,579 @@
+            spear.log(
+                "WARNING: Unexpected static mesh path: ",
+                get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_static_mesh(static_mesh), ")")
+            return
+        if not static_mesh_path[7].startswith("SM_"):
+            spear.log(
+                "WARNING: Unexpected static mesh path: ",
+                get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_static_mesh(static_mesh), ")")
+            return
+
+        #
+        # Each material is kept here in a directory named according to the material's
+        # globally unique ID.
+        #
+
+        materials = component.get_editor_property("override_materials")
+        for material in materials:
+            if material is None:
+                spear.log(
+                    "WARNING: Unexpected material: ",
+                    get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                return
+
+            material_path = pathlib.PurePosixPath(material.get_path_name()).parts
+            if len(material_path) < 4:
+                spear.log(
+                    "WARNING: Unexpected material path: ",
+                    get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                return
+
+            if material_path[0:5] != ("/", "Game", "Scenes", level, "Materials") and \
+                    material_path[0:3] != ("/", "Game", "Megascans") and \
+                    material_path[0:4] != ("/", "Game", "Kujiale", "Materials"):
+                spear.log(
+                    "WARNING: Unexpected material path: ",
+                    get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                return
+
+            base_material = material.get_base_material()
+            if base_material.get_name() == "M_BaseMaterial":
+                if material_path[-1] != "M_BaseMaterial.M_BaseMaterial" and not material_path[-1].startswith("MI_"):
+                    spear.log(
+                        "WARNING: Unexpected material path: ",
+                        get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                    return
+
+            if base_material.get_name() == "M_TranslucentMaterial":
+                if material_path[-1] != "M_TranslucentMaterial.M_TranslucentMaterial" and not material_path[-1].startswith("MI_"):
+                    spear.log(
+                        "WARNING: Unexpected material path: ",
+                        get_debug_string_actor(actor), ".", get_debug_string_component(component), " (", get_debug_string_material(material), ")")
+                    return
+
+
```

Comment:
Group `get_parent_components - get_debug_string_actor` with only one line of whitespace in between them because they're all conceptually related, i.e., they're all low-level helper functions.